docs/source/kedro_project_setup/settings.md

```diff
@@ -0,0 +1,98 @@
+# Project settings
+
+A Kedro project's `settings.py` file contains the settings for the project, such as library component registration, custom hooks registration, etc. This document explains how settings work and which settings are available.
+
+## Default settings
+A Kedro settings file doesn’t have to define any settings if it doesn’t need to. Each setting has a sensible default value. These defaults live in the module `src/<project_name>/settings.py`.
+
+## Available settings
+
+#### HOOKS
+Default: `tuple()`
+
+Instantiate and list your project hooks.
+
+```python
+HOOKS = (ProjectHooks(),)
+```
+
+#### DISABLE_HOOKS_FOR_PLUGINS
+Default: `tuple()`
+
+List the installed plugins for which to disable auto-registry.
+
+```python
+DISABLE_HOOKS_FOR_PLUGINS = ("kedro-viz",)
+```
+
+#### SESSION_STORE_CLASS
+Default: `kedro.framework.session.session.BaseSessionStore`
+
+Define where to store data from a KedroSession.
+
+```python
+from kedro.framework.session.store import ShelveStore
+
+SESSION_STORE_CLASS = ShelveStore
+```
+
+#### SESSION_STORE_ARGS
+Default: { } (empty dictionary)
+
+Define keyword arguments to be passed to `SESSION_STORE_CLASS` constructor.
+
+```python
+SESSION_STORE_ARGS = {"path": "./sessions"}
+```
+
+#### CONTEXT_CLASS
+Default: `kedro.framework.context.KedroContext`
+
+Define custom context class.
+```python
+from myproject import CustomContext
+
+CONTEXT_CLASS = CustomContext
+```
+
+#### CONF_SOURCE
+Define the configuration folder. Defaults to `conf`
+
+```python
+CONF_SOURCE = "conf"
+```
+
+#### CONFIG_LOADER_CLASS
+Default: `kedro.config.ConfigLoader`
+
+Define the project ConfigLoader class.
+
+```python
+from kedro.config import TemplatedConfigLoader
+
+CONFIG_LOADER_CLASS = TemplatedConfigLoader
+```
+
+#### CONFIG_LOADER_ARGS
+Default: { } (empty dictionary)
+
+Define keyword arguments to be passed to `CONFIG_LOADER_CLASS` constructor. These kwargs depend on the `ConfigLoader` class implementation.
+
+```python
+CONFIG_LOADER_ARGS = {
+    "globals_pattern": "*globals.yml",
+    "base_env": "base",
+    "default_run_env": "local",
+}
+```
+
+#### DATA_CATALOG_CLASS
+Default: `kedro.io.DataCatalog`
+
+Define the project `DataCatalog` class.
+
+```python
+from myproject import CustomCatalog
+
+DATA_CATALOG_CLASS = CustomCatalog
+```
```

Comment:
General comment on this new page: I would keep it simple and leave out all these snippets of python. Since they're just copy and pasted from settings.py I don't think really add anything here. In my opinion the value of this page would be to summarise/give a slightly different perspective on what's available in settings.py rather than just reproducing that file in markdown. Since, unlike the actual settings.py file, it's also clickable it would be nice to directly link to other pages in the documentation here also.

I would suggest doing this as a table like this:
```
```eval_rst
+-----------------------------------+---------------+----------------------------------------------------------------+
| Setting                           | Default value | Use                                                            |
+===================================+===============+================================================================+
| :code:                            | :code:        | Instantiate and list your `project hooks <../extend_kedro/hooks.md>`_.                       |
| `HOOKS`                           | `tuple()`     |                                                                |
+-----------------------------------+---------------+----------------------------------------------------------------+
| :code:`DISABLE_HOOKS_FOR_PLUGINS` | :code:        | List the installed plugins for which to disable auto-registry. |
|                                   | `tuple()`     |                                                                |
+-----------------------------------+---------------+----------------------------------------------------------------+
|                                   |               |                                                                |
+-----------------------------------+---------------+----------------------------------------------------------------+
```

Reply:
I like the idea of structuring it as a table. I initially took inspiration from [Django](https://docs.djangoproject.com/en/3.2/ref/settings/), but the table will look nicer. In general the page is very much still in draft mode, but I wanted to start with something.